core/container/build_registry.go

```diff
@@ -35,6 +35,19 @@ func (br *BuildRegistry) BuildStatus(ccid string) (*BuildStatus, bool) {
 	return bs, ok
 }
 
+// ResetBuildStatus returns a new BuildStatus for the ccid, and false,
+// indicating that this build status is their responsibility. The caller
+// must call Notify with the error (or nil) upon completion.
```

Comment:
I would probably note that it's the caller's responsibility to ensure that `ResetBuildStatus` is only called when an existing build status is done.  This means that the caller must use external locking to ensure that the build status is not reset between the check that the status is done, and when the build status is reset.

Reply:
Updated comment.